include/bluetooth/gatt.h

```diff
@@ -1217,10 +1217,11 @@ struct bt_gatt_exchange_params {
  *  @note Shall only be used once per connection.
  *
  *  The Response comes in callback @p params->func. The callback is run from
- *  the BT RX thread. @p params must remain valid until start of callback.
+ *  the BT RX thread or System work queue when @kconfig{CONFIG_BT_RECV_IS_RX_THREAD}
+ *  not selected. @p params must remain valid until start of callback.
  *
  *  This function will block while the ATT request queue is full, except when
- *  called from the BT RX thread, as this would cause a deadlock.
+ *  called from the Bluetooth callback, as this would cause a deadlock.
```

Comment:
Applies multiple places:
```suggestion
 *  called from a Bluetooth callback, as this would cause a deadlock.
```
or
```suggestion
 *  called from the Bluetooth callback thread, as this would cause a deadlock.
```